src/ferc_xbrl_extractor/datapackage.py

```diff
@@ -332,6 +334,36 @@ def get_period_type(self):
         period_type = "instant" if "date" in self.schema_.primary_key else "duration"
         return period_type
 
+    def merge_resources(self, other: "Resource", other_version: str) -> "Resource":
```

Comment:
This is attempting to merge the schema of the same table from two different versions of a taxonomy. It first looks at primary key and raises an error if it's changed. Right now this never occurs, so it's not a problem, but if future versions change a primary key of a table, we'll need to decide how to handle this. After checking the primary key, it will look for any new or missing columns. If there are new columns, it will append these. Changes in the set of columns will lead to Nulls in the extracted table. So far there haven't been many changes in the column set, fortunately.

Reply:
This is similar the actual behavior of the DBF derived data, in which we verified that as the DB schema evolved over the years, it was only ever additive -- new tables, and new columns within existing tables, but never the removal of a table of columns, which meant that we could always just use the latest published database to define a schema which was able to contain all years of data. But here you don't have that temporal dependency.

Reply:
Can we note this in the docstring of this method? Seems helpful to note what the underlying merge behavior is.

Reply:
@zaneselvans that makes sense, Dazhong and I have talked about making the extractor produce tidy tables, which would allow us to handle these changes in a much cleaner way, but decided to hold off for now.

@e-belfer yes! Sorry forgot to really go back and improve the documentation, but should be better now.